src/main/java/com/rcv/Main.java

```diff
@@ -170,102 +170,52 @@ static void executeTabulation(ElectionConfig config) {
 
     // castVoteRecords will contain all cast vote records parsed by the reader
     List<CastVoteRecord> castVoteRecords = new ArrayList<>();
-    // how many sources had fatal problems?
-    int numInvalidSources = 0;
-
-    if (config.rawConfig.cvrFileSources == null || config.rawConfig.cvrFileSources.isEmpty()) {
-      Logger.tabulationLog(Level.SEVERE, "Config doesn't contain any CVR input files.");
-    } else {
-      // At each iteration of the following loop, we add records from another source file.
-      // source: index over config sources
-      for (RawElectionConfig.CVRSource source : config.rawConfig.cvrFileSources) {
-        Logger.tabulationLog(
-            Level.INFO,
-            "Reading CVR file: %s (provider: %s)",
-            source.getFilePath(),
-            source.getProvider());
-
-        // did we encounter a fatal problem for this source?
-        boolean encounteredProblemForThisSource = false;
-
-        if (source.getFilePath() == null || source.getFilePath().isEmpty()) {
-          Logger.tabulationLog(Level.SEVERE, "Invalid source file: missing filePath");
-          encounteredProblemForThisSource = true;
-        }
+    // did we encounter a fatal problem for this source?
+    boolean encounteredSourceProblem = false;
+
+    // At each iteration of the following loop, we add records from another source file.
+    // source: index over config sources
+    for (RawElectionConfig.CVRSource source : config.rawConfig.cvrFileSources) {
+      String logMessage = String.format("Reading CVR file: %s", source.getFilePath());
```

Comment:
prefer `StringBuilder`

Reply:
Yeah, I looked this up recently, and it does seem like [StringBuilder is better-performing](https://stackoverflow.com/questions/44117788/performance-between-string-format-and-stringbuilder). I saw a precedent in the Logger class for using String.format though and went with that. Might be worth filing an issue to make a pass over everything and converting?

Reply:
Makes sense for the Logger class at least.  I agree with the link there's places where we should keep String.format for readability.  I'll do that for the logger when I fix the other errors.

Reply:
Sorry, let me clarify. I'm fine with using `String.format` for the sake of readability. What I want to avoid is multiple statements appending to a String variable, especially in a loop, because that means we're actually creating a new String object each time.